### PR TITLE
Corrects confusing error message when manifest not found

### PR DESCRIPTION
**Pull request by samsalisbury**
Resolves SOUS-1020; includes #804; See https://github.com/opentable/sous/pull/805/files#diff-91432d762aa99853969261d3158edbb4R15 for test case covering the bug report, and https://github.com/opentable/sous/pull/805/commits/e13914ca4845dc165bbe1da6887be4c8e7f07aac for the resolution.

### CODE REVIEW

**Comment by sous-machine-account** [comment]
# LGTM!

No errors, no warnings. :shipit:


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/resolvefilter.go [hunk 1]
```diff
@@ -156,15 +156,14 @@ func (rf *ResolveFilter) DeploymentID(mid ManifestID) (DeploymentID, error) {
...
 func (rf *ResolveFilter) String() string {
-
 	return fmt.Sprintf(
 		"<cluster:%s repo:%s offset:%s flavor:%s tag:%s revision:%s>",
 		rf.Cluster.ValueOr("*"),
 		rf.Repo.ValueOr("*"),
-		rf.Revision.ValueOr("*"),
-		rf.Flavor.ValueOr("*"),
 		rf.Offset.ValueOr("*"),
+		rf.Flavor.ValueOr("*"),
```
Comment (nyarly) [inline lib/resolvefilter.go]: Unit tests for this would be rad.

Comment (samsalisbury) [reply lib/resolvefilter.go]: They would be... man it's fiddly to test. I wish I could take the address of a string literal here...

Comment (samsalisbury) [reply lib/resolvefilter.go]: @nyarly test added

**Pull request merged**